simpeg/electromagnetics/natural_source/receivers.py

```diff
@@ -1,42 +1,122 @@
-from ...utils.code_utils import validate_string
-
+from ...utils.code_utils import (
+    deprecate_class,
+    validate_string,
+    validate_ndarray_with_shape,
+)
+import warnings
 import numpy as np
 from scipy.constants import mu_0
-
 from ...survey import BaseRx
 
 
 def _alpha(src):
     return 1 / (2 * np.pi * mu_0 * src.frequency)
 
 
-class PointNaturalSource(BaseRx):
-    """Point receiver class for magnetotelluric simulations.
+# THIS SHOULD BE HIDDEN METHOD
+def _getP(rx, mesh, projected_grid, field="e", is_tipper_bs=False):
+    """Projection matrix from discrete field solution to measurement locations.
 
-    Assumes that the data locations are standard xyz coordinates;
-    i.e. (x,y,z) is (Easting, Northing, up).
+    Note projection matrices are stored as a dictionary listed by meshes.
+
+    Parameters
+    ----------
+    rx : .natural_source.receivers.PointImpedance
+        a NSEM receiver
+    mesh : discretize.base.BaseMesh
+        The mesh on which the discrete set of equations is solved.
+    projected_grid : str
+        Define what part of the mesh (i.e. edges, faces, centers, nodes) to
+        project from. Must be one of::
+
+            'Ex', 'edges_x'           -> x-component of field defined on x edges
+            'Ey', 'edges_y'           -> y-component of field defined on y edges
+            'Ez', 'edges_z'           -> z-component of field defined on z edges
+            'Fx', 'faces_x'           -> x-component of field defined on x faces
+            'Fy', 'faces_y'           -> y-component of field defined on y faces
+            'Fz', 'faces_z'           -> z-component of field defined on z faces
+            'N', 'nodes'              -> scalar field defined on nodes
+            'CC', 'cell_centers'      -> scalar field defined on cell centers
+            'CCVx', 'cell_centers_x'  -> x-component of vector field defined on cell centers
+            'CCVy', 'cell_centers_y'  -> y-component of vector field defined on cell centers
+            'CCVz', 'cell_centers_z'  -> z-component of vector field defined on cell centers
+
+    field : {"e", "h"}
+        Whether to project electric or magnetic fields from the mesh.
+    is_tipper_bs : bool
+        Whether the projection is for a remote base station location where horizontal
+        magnetic fields are measured. Ensures stash names for projection matrices
+        aren't reused.
+    """
+    # Inherited method from BaseRx projecting to self.locations
+    if mesh.dim < 3:
+        return rx.getP(mesh, projected_grid)
+
+    if is_tipper_bs:
+        stash_name = projected_grid + "_bs"
+    else:
+        stash_name = projected_grid
```

Comment:
This happens behind the scenes. So I don't think it's too important. I decided to make this function hidden so that I could use it universally for all receiver classes.  The important part is ensuring you can simulate AFMAG data or ZTEM data. I decided to make this function hidden so that I could use it universally for all receiver classes.

Reply:
Makes sense!